config/samples/dspipelines.io_v1alpha1_dspipeline.yaml

```diff
@@ -2,44 +2,105 @@ apiVersion: dspipelines.opendatahub.io/v1alpha1
 kind: DSPipeline
 metadata:
   name: sample
+  namespace: data-science-project
 spec:
   apiServer:
+    deploy: true
     # optional (default is: ds-pipeline-artifact-script-${metadata.name} / artifact_script)
     artifactScriptConfigMap:
       name: ds-pipeline-artifact-script-sample
       key: "artifact_script"
     apiServerImage: quay.io/modh/odh-ml-pipelines-api-server-container:v1.18.0-8
     artifactImage: quay.io/modh/odh-ml-pipelines-artifact-manager-container:v1.18.0-8
+    dBConfigConMaxLifetimeSec: 120
+    CollectMetricsFlag: true
+    resources:
+      requests:
+        cpu: 250m
+        memory: 500Mi
+      limits:
+        cpu: 500m
+        memory: 1Gi
   persistentAgent:
+    deploy: true
     image: quay.io/modh/odh-ml-pipelines-persistenceagent-container:v1.18.0-8
     pipelineAPIServerName: apiserver
+    numWorkers: 2  # Number of worker for sync job.
+    resources:
+      requests:
+        cpu: 120m
+        memory: 500Mi
+      limits:
+        cpu: 250m
+        memory: 1Gi
   scheduledWorkflow:
+    deploy: true
     image: quay.io/modh/odh-ml-pipelines-scheduledworkflow-container:v1.18.0-8
+    resources:
+      requests:
+        cpu: 120m
+        memory: 100Mi
+      limits:
+        cpu: 250m
+        memory: 250Mi
   viewerCRD:
+    deploy: true
     image: quay.io/modh/odh-ml-pipelines-viewercontroller-container:v1.18.0-8
+    resources:
+      requests:
+        cpu: 120m
+        memory: 100Mi
+      limits:
+        cpu: 250m
+        memory: 500Mi
   mlpipelineUI:
+    deploy: true
     image: quay.io/opendatahub/odh-ml-pipelines-frontend-container:beta-ui
     configMap: ds-pipeline-ui-configmap
+    resources:
+      limits:
+        cpu: 100m
+        memory: 256Mi
+      requests:
+        cpu: 100m
+        memory: 256Mi
   database:
     mariaDB:   # mutually exclusive with custom
+      deploy: true
       image: registry.redhat.io/rhel8/mariadb-103:1-188
       username: mlpipeline
       pipelineDBName: randomDBName
+      PVCSize: 20Gi
+      resources:
+        requests:
+          cpu: 300m
+          memory: 800Mi
+        limits:
+          cpu: "1"
+          memory: 1Gi
       passwordSecret:
         name: ds-pipelines-db-sample
         key: password
-    # customDB:
-    #   host: mysql:3306
-    #   port: "8888"
-    #   username: root
-    #   pipelineDBName: randomDBName
-    #   passwordSecret:
-    #     name: somesecret
-    #     key: somekey
-  storage:
+#   customDB:
```

Comment:
does it make sense to have a `type: maria|custom|mysql|etc` field instead of making these mutex-able db items?

Reply:
to me it makes sense to have custom (operator deploys a db for us) separate from maria|postgresql|mysql (we tell operator to use an existing db), since fields are different, and the handling operator logic is different (we supply our own db vs operator sets it up for us)

Reply:
if the sub-fields are different, I think it's more typical to have separate objects

an example that jumped to mind is the openshift install config where you set platform-specific stuff:

```yaml
compute:
- name: worker
  platform:
    aws:
      amiID: ami-06c4d345f7c207239 
      type: m5.4xlarge
```

aws, gcp, etc are all similar but slightly different fields, i.e. `amiID` means nothing in gcp

so gcp has its own object that doesn't include an `amiID`

Reply:
As per in call discussion, we have settled for having external db connection as a separate dict than mariadb deployment dict. I've updated the cr to reflect this change.